docs/reference/auth/webhooks.rst

```diff
@@ -0,0 +1,256 @@
+.. _ref_auth_webhooks:
+
+========
+Webhooks
+========
+
+The auth extension supports sending webhooks for a variety of auth events. You can use these webhooks to, for instance, send a fully customized email for email verification, or password reset instead of our built-in email verification and password reset emails. You could also use them to trigger analytics events, start an email drip campaign, create an audit log, or trigger other side effects in your application.
+
+If you are using Webhooks to send emails, be sure to not also configure an SMTP provider otherwise we will send the email via SMTP and also send the webhook which will trigger your custom email sending behavior.
+
+.. warning::
+
+  We send webhooks with no durability or reliability guarantees, so you should always provide a mechanism for retrying delivery of any critical events, such as email verification and password reset. We detail how to resend these events in the relevant sections on the various authentication flows.
+
+Configuration
+=============
+
+You can configure webhooks with the UI or via query.
+
+.. code-block:: edgeql
+
+    configure current branch insert
+      ext::auth::WebhookConfig {
+        url := 'https://example.com/auth/webhook',
+        events := {
+          ext::auth::WebhookEvent.EmailVerificationRequested,
+          ext::auth::WebhookEvent.PasswordResetRequested,
+        },
+        # Optional, only needed if you want to verify the webhook request
+        signing_secret_key := '1234567890',
+      };
+
+.. note::
+
+  URLs must be unique across all webhooks configured for each branch. If you want to send multiple events to the same URL, you can do so by adding multiple ``ext::auth::WebhookEvent`` values to the ``events`` set.
```

Comment:
Maybe we can add "as demonstrated in the example above"?

Reply:
Oh good call. I think maybe I'll just move this whole note into the paragraph that precedes the example so it's clear what to look out for when reading the example.

Reply:
@beerose ⚡ [`f0134cf` (#8548)](https://github.com/geldata/gel/pull/8548/commits/f0134cf26e0c171285437f48573894cc419a3e7c)